lib/vtls/schannel.c

```diff
@@ -405,6 +405,362 @@ get_cert_location(TCHAR *path, DWORD *store_name, TCHAR **store_path,
 }
 #endif
 
+static CURLcode
+schannel_acquire_credential_handle(struct Curl_easy *data,
+                                   struct connectdata *conn,
+                                   int sockindex)
+{
```

Comment:
This function is very complicated with large amount of error code paths and freeing various buffers and resources in them. Since it is so complicated it is very difficult to keep track of all resources, not alone modify the function later, if needed. It could perhaps use refactoring to smaller functions, or some some more robust way of handling errors.

Reply:
I totally agree with @piru

Reply:
that code is yanked out of schannel_connect_step1, which already existed. I believe moving it out of schannel_connect_step1 reduces the complexity. I can see that the function can further split 2-3 functions, but that can come later as refactoring change, if necessary.

Reply:
Maybe the life-time of several variables which content lifes on the heap might be a good indicator to decide where to split